src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_CAPACITY = Integer.MAX_VALUE;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        raiseIfRequirement();
```

Comment:
Let's better name it in that way:
```suggestion
        growIfNeeded();
```

[Reference](https://www.jetbrains.com/help/idea/rename-refactorings.html#invoke-rename-refactoring)